joblib/numpy_pickle_utils.py

```diff
@@ -211,11 +211,11 @@ def _read_fileobject(fileobj, filename, mmap_mode=None):
                 # binary mode. In this case, we pass the filename.
                 yield _buffered_read_file(bz2.BZ2File(fileobj.name, 'rb'))
         elif (compressor == 'lzma' or compressor == 'xz'):
-            if lzma is not None:
+            if PY3_OR_LATER and lzma is not None:
```

Comment:
Maybe add a comment here saying that there is a pylzma module for Python 2.7 that is unfortunately not exactly similar as the Python 3 lzma so we only support lzma decompression for Python 3, or something better.